### PR TITLE
User/ohbitton/add file options

### PR DESCRIPTION
**Pull request by ohadbitt**
### Added
RawDataSize, CompressionType file options

copilot:all

### CODE REVIEW

File: kusto/ingest/internal/queued/queued.go [hunk 1]
```diff
@@ -396,6 +380,75 @@ func (i *Ingestion) localToBlob(ctx context.Context, from string, client *azblob
 	return fullUrl(client, container, blobName), stat.Size(), nil
 }
 
+// CompressionDiscovery looks at the file extension. If it is one we support, we return that
+// CompressionType that represents that value. Otherwise we return CTNone to indicate that the
+// file should not be compressed.
+func CompressionDiscovery(fName string) source.CompressionType {
```
Comment (AsafMah) [inline kusto/ingest/internal/queued/queued.go:386]: Shouldn't this be under the new namespace?

File: kusto/test/etoe/etoe_test.go [hunk 2]
```diff
@@ -2258,6 +2259,7 @@ func createStringyLogsData() string {
 
 func executeCommands(client *kusto.Client, database string, commandsToRun ...kusto.Statement) error {
 	for _, cmd := range commandsToRun {
+		time.Sleep(100 * time.Millisecond)
```
Comment (AsafMah) [inline kusto/test/etoe/etoe_test.go:2262]: ?

File: kusto/test/etoe/ingestion_status_e2e_test.go [hunk 3]
```diff
@@ -240,8 +240,8 @@ func TestIngestionStatus(t *testing.T) {
 		defer cancel()
 
 		f, err := os.Open(csvFile)
-		defer f.Close()
 		require.NoError(t, err)
+		defer f.Close()
```
Comment (AsafMah) [inline kusto/test/etoe/ingestion_status_e2e_test.go:244]: why?

File: kusto/ingest/internal/queued/queued_test.go [hunk 4]
```diff
@@ -56,13 +58,13 @@ func TestCompressionDiscovery(t *testing.T) {
 
 	tests := []struct {
 		input string
-		want  properties.CompressionType
+		want  source.CompressionType
```
Comment (AsafMah) [inline kusto/ingest/internal/queued/queued_test.go:61]: No new tests?

File: kusto/ingest/source/source.go [hunk 5]
```diff
@@ -0,0 +1,27 @@
+package source
```
Comment (AsafMah) [inline kusto/ingest/source/source.go:1]: Add to changelog.md

**Review by AsafMah — CHANGES_REQUESTED** [review]

File: kusto/test/etoe/etoe_test.go (hunk 2 above)
Comment (ohadbitt) [reply kusto/test/etoe/etoe_test.go:2262]: can remove - again - same problem i have with testing against my cluster - can you remind  me what to change in the cluster policies to achive it

File: kusto/test/etoe/ingestion_status_e2e_test.go (hunk 3 above)
Comment (ohadbitt) [reply kusto/test/etoe/ingestion_status_e2e_test.go:244]: Don't recall but why not

File: kusto/ingest/internal/queued/queued_test.go (hunk 4 above)
Comment (ohadbitt) [reply kusto/ingest/internal/queued/queued_test.go]: now added

File: kusto/ingest/ingestoptions/ingestoptions.go [hunk 6]
```diff
@@ -0,0 +1,154 @@
...
+	TSVE DataFormat = 14
+	// TXT is a text file with lines delimited by "\n".
+	TXT DataFormat = 15
+	// W3CLogFile indicates the source is encoded using W3C Extended Log File format.
+	W3CLogFile DataFormat = 16
+	// SingleJSON indicates the source is a single JSON value -- newlines are regular whitespace.
+	SingleJSON DataFormat = 17
+)
+
+type dfDescriptor struct {
```
Comment (AsafMah) [inline kusto/ingest/ingestoptions/ingestoptions.go:83]: This should not be here, right? it's before your change and it's not used

**Review by AsafMah — APPROVED** [review]

**Pull request merged**